src/imitation/algorithms/preference_comparisons.py

```diff
@@ -765,10 +780,17 @@ def forward(
         predictions = (probs > 0.5).float()
         preferences_th = th.as_tensor(preferences, dtype=th.float32)
         ground_truth = (preferences_th > 0.5).float()
-        accuracy = (predictions == ground_truth).float().mean()
+        metrics = {}
+        metrics["accuracy"] = (predictions == ground_truth).float().mean()
+        if gt_reward_available:
+            metrics["gt_reward_loss"] = th.nn.functional.binary_cross_entropy(
+                gt_probs,
+                preferences_th,
+            )
+
         return LossAndMetrics(
             loss=th.nn.functional.binary_cross_entropy(probs, preferences_th),
-            metrics={"accuracy": accuracy.detach().cpu()},
+            metrics=metrics,
```

Comment:
Previously we had `.detatch().cpu()`, now we're just returning raw tensors I think. Is this supported?

Reply:
This could cause crashes if we are training on the GPU or lead to a memory leak. Good catch. I just fixed it.